src/main/java/com/gluonhq/substrate/target/DarwinTargetConfiguration.java

```diff
@@ -56,27 +70,49 @@ String getAdditionalSourceFileLocation() {
         return Arrays.asList("AppDelegate.m", "launcher.c");
     }
 
+    @Override
+    boolean useGraalVMJavaStaticLibraries() {
+        return true;
+    }
+
     @Override
     List<String> getTargetSpecificLinkFlags(boolean useJavaFX, boolean usePrismSW) {
-        if (!useJavaFX) {
-            return darwinLibs;
+        List<String> linkFlags = new ArrayList<>();
+
+        linkFlags.addAll(asListOfLibraryLinkFlags(javaDarwinLibs));
+        if (useJavaFX) {
+            linkFlags.addAll(asListOfLibraryLinkFlags(javaFxDarwinLibs));
+        }
+
+        linkFlags.addAll(asListOfFrameworkLinkFlags(javaDarwinFrameworks));
+        if (useJavaFX) {
+            linkFlags.addAll(asListOfFrameworkLinkFlags(javaFxDarwinFrameworks));
         }
-        String libPath = "-Wl,-force_load," + projectConfiguration.getJavafxStaticLibsPath() + "/";
-        List<String> answer = new ArrayList<>(Arrays.asList(
-                libPath + "libprism_es2.a", libPath + "libglass.a",
-                libPath + "libjavafx_font.a", libPath + "libjavafx_iio.a"));
-        if (usePrismSW) {
-            answer.add(libPath + "libprism_sw.a");
+
+        if (useJavaFX) {
+            List<String> javafxLibs = new ArrayList<>(staticJavaFxLibs);
+            if (usePrismSW) {
+                javafxLibs.add("prism_sw");
+            } else {
+                javafxLibs.add("prism_es2");
```

Comment:
should prism_es2 be added even if sw is allowed? In that case, you build an executable that will use es2 if possible, but falls back to sw if not possible

Reply:
Good point. My understanding of `usePrismSW` was that it meant to use sw pipeline instead of the hardware accelerated version. Maybe it should be renamed to mean that it will add the sw pipeline on top of the hardware pipeline, e.g. `includePrismSW`.?